gui/unknown-game-dialog.cpp

```diff
@@ -122,7 +122,7 @@ void UnknownGameDialog::rebuild() {
 Common::U32String UnknownGameDialog::encodeUrlString(const Common::U32String &string) {
 	Common::U32String encoded;
 	for (uint i = 0 ; i < string.size() ; ++i) {
-		char c = string[i];
+		uint32 c = string[i];
```

Comment:
Encoding non-ASCII characters that way probably doesn't work? I think it needs to be converted to UTF-8 first.

Reply:
`Common::U32String report = generateUnknownGameReport(_detectedGame, false, false);`
I didn't think it mattered because these strings are ASCII only (the above function, second argument is for `translate`, and since it's `false`, the returned string is ascii (https://github.com/aryanrawlani28/scummvm/blob/3fe5e20b1482a7b2e84c6489eb65b284418e9a53/engines/game.cpp#L168)
I changed it just to match consistency, I'll change the function to accept a `Common::String` instead.